app/src/main/java/org/fossasia/phimpme/utilities/ImgurRetrofitClient.java

```diff
@@ -0,0 +1,40 @@
+package org.fossasia.phimpme.utilities;
+
+import okhttp3.OkHttpClient;
+import okhttp3.logging.HttpLoggingInterceptor;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+/** Created by @codedsun on 19/Oct/2019 */
+
+// Retrofit client used for Imgur
+public class ImgurRetrofitClient {
+
+  private Retrofit retrofit;
+  private final String IMGUR_BASE_URL = "https://api.imgur.com/3/";
+  private static ImgurRetrofitClient imgurRetrofitClient = null;
+
+  private ImgurRetrofitClient() {
+    retrofit =
+        new Retrofit.Builder()
+            .baseUrl(IMGUR_BASE_URL)
+            .client(httpClient().build())
+            .addConverterFactory(GsonConverterFactory.create())
+            .build();
+  }
+
+  private static OkHttpClient.Builder httpClient() {
+    OkHttpClient.Builder httpClient = new OkHttpClient.Builder();
+    HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
+    interceptor.level(HttpLoggingInterceptor.Level.BODY);
+    httpClient.interceptors().add(interceptor);
+    return httpClient;
+  }
+
+  public static Retrofit getImgurRetrofitClient() {
+    if (imgurRetrofitClient == null) {
+      imgurRetrofitClient = new ImgurRetrofitClient();
+    }
+    return imgurRetrofitClient.retrofit;
+  }
+}
```

Comment:
No need to create an entire class and factory just for another Base URL

Reply:
means I should shift this code to activity?

Reply:
I have renamed the earlier class from retrofitClient to ImgurRetrofitClient, It follows singleton pattern, that's why a seperat class is created and also you suggested 

>Then you have to create another retrofit instance in any case! It has nothing to do with the fact that you are creating the service in method each time, instead of 1 time. You may use singleton or not, that is your choice. Both of those approaches will not save you from having to create another retrofit instance for pinterest upload

Reply:
@iamareebjamal Please guide me what should be done?